src/PaymentChannel.js

```diff
@@ -0,0 +1,114 @@
+import BigNumber from 'bignumber.js';
+
+import paymentChannelStateMessages from './payment_channel_state_service_pb';
+
+export default class PaymentChannel {
+  constructor(channelId, web3, account, paymentChannelStateServiceClient, mpeContract) {
+    this._channelId = channelId;
+    this._web3 = web3;
+    this._account = account;
+    this._mpeContract = mpeContract;
+    this._paymentChannelStateServiceClient = paymentChannelStateServiceClient;
+    this._state = {
+      nonce: new BigNumber(0),
+      lastSignedAmount: new BigNumber(0),
+    };
+  }
+
+  get channelId() {
+    return this._channelId;
+  }
+
+  get nonce() {
+    return this.state.nonce;
+  }
+
+  get lastSignedAmount() {
+    return this.state.lastSignedAmount;
+  }
+
+  get state() {
+    return this._state;
+  }
+
+  updateState(gstate) {
+    this._state = {
+      ...this._state,
+      ...state,
+    }
+  }
+
+  hasSufficientFunds(amount) {
+    return this.state.availableAmount >= amount;
+  }
+
+  isValid(expiry) {
+    return this.state.expiration > expiry;
+  }
+
+  async addFunds(amount) {
+    return this._mpeContract.channelAddFunds(this._account, this.channelId, amount);
+  }
+
+  async extendExpiration(expiration) {
+    return this._mpeContract.channelExtend(this._account, this.channelId, expiration);
+  }
+
+  async extendAndAddFunds(expiration, amount) {
+    return this._mpeContract.channelExtendAndAddFunds(this._account, this.channelId, expiration, amount);
+  }
+
+  async syncState() {
+    const latestChannelInfoOnBlockchain = await this._mpeContract.channels(this.channelId);
+    const currentState = await this._currentChannelState();
+    const { lastSignedAmount, nonce: currentNonce } = currentState;
+    const { nonce, expiration, value: totalAmount } = latestChannelInfoOnBlockchain;
+    const availableAmount = totalAmount - lastSignedAmount;
+    this._state = {
+      nonce,
+      currentNonce,
+      expiration,
+      totalAmount,
+      lastSignedAmount,
+      availableAmount,
+    };
+    return this;
+  }
+
+  async _currentChannelState() {
```

Comment:
We can consider refactoring the code to sign into its own utility method. See https://github.com/singnet/snet-dapp/blob/dac23c85a8666bcaaaf7eac4d9ef5e476431b3b2/src/components/JobDetails.js#L148 as a possible example. This way the signing code can be reused across

Reply:
Got it. Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/1/commits/adec39ac3615b848e67350013b240d34af86a6fb)